ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -240,7 +243,7 @@ module Formatter = struct
     | Result.Error e ->
       let message = Fmt.message e in
       let error = jsonrpc_error e in
-      let msg = ShowMessageParams.create ~message ~type_:Error in
+      let msg = ShowMessageParams.create ~message ~type_:Warning in
```

Comment:
Is this change done on purpose? If so, I don't see the reasoning. A failure to format is an error not a warning.

Reply:
Not on purpose, but would be great to have this change (maybe in a separate PR). Reasoning: 

When a user saves the file, they get this scary _error_ pop-up

![image](https://user-images.githubusercontent.com/16353531/125160451-5c035b80-e196-11eb-9bb1-357ae3045c14.png)

And even though I KNOW (at this point) the file is saved, every time it feels like the file wasn't saved because there is an **error**. I'm sure there are others who feel same way. 

A failure to format is an error, but the error being shown on a file save seems bothering (at least to me)

Reply:
Alright, I don't see much harm in making it a warning. We do need make this error message less noisy though.